tests/test_scheduler.py

```
@@ -0,0 +1,527 @@
+        # Train 2 trees
+        for _ in range(2):
+            y_pred = model(X_tensor, requires_grad=True)
+            loss = 0.5 * th.mean((y_pred - y_tensor.squeeze()) ** 2)
+            loss.backward()
+            model.step()
+        
+        # Get predictions from each tree individually
+        pred_tree0 = self._get_tree_predictions(model, 'cuda', 0, 1)
+        pred_tree1 = self._get_tree_predictions(model, 'cuda', 1, 2)
+        
+        print(f"\nGPU Linear Scheduler Test:")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)